add ECC helpers to get size and id from curve name

cconlon opened the pull request:
This PR adds two ECC helper functions:

1) Get ECC curve size from curve name
`int wc_ecc_get_curve_size_from_name(const char* curveName);`

2) Get ECC curve ID from curve name
`int wc_ecc_get_curve_id_from_name(const char* curveName);`

Corresponding unit tests are also added.

dgarske commented on wolfcrypt/src/ecc.c:
```diff
@@ -2490,6 +2490,65 @@ int wc_ecc_get_curve_size_from_id(int curve_id)
+    /* find curve from name */
+    for (x = 0; ecc_sets[x].size != 0; x++) {
+        if (XSTRNCMP(ecc_sets[x].name, curveName,
+                    XSTRLEN(curveName)) == 0) {
```
Hey Chris, just a couple thoughts:

1. Recommend doing XSTRLEN() above as local variable, so it doesn't have to be done multiple times.
2. Recommend adding a case insensitive strnicmp support to our library and using that here.
3. We probably need to support "PRIME256V1" as a match for "SECP256R1".
4. To reduce duplicate code I'd add a third public function to get the ecc_set index for a curveName. Then call it in these two new functions.

cconlon commented:
Thanks for the feedback David, changes made:

1. Only do XSTRLEN() once, as suggested
2. Added case insensitive compare using XSTRNCASECMP
3. Added third `wc_ecc_get_curve_idx_from_name()` and used in other functions

I didn't handle mapping of curves with same sizes to each other.  I wasn't sure if the additional string compare operations and code maintenance would warrant it or not.

Thanks!
Chris
